### ISSUE TITLE
I think should throw an error when dispatch a unkonwn mutation.

### ISSUE THREAD

**Issue by wszgxa** [vuejs/vuex#538]
in `1.0.0-rc.2`, it didn't throw an error and change the other state.

in devtool:

![](http://7fvhwe.com1.z0.glb.clouddn.com/%E5%B1%8F%E5%B9%95%E5%BF%AB%E7%85%A7%202016-12-22%20%E4%B8%8A%E5%8D%8810.45.29.png)

**Comment by ktsn** [comment]
I'm not sure what the point your suggestion is. Could you explain more clearly why throwing error is needed?
Because Vuex v1.0 already provide a warning when a unknown mutation is dispatched.

**Comment by wszgxa** [comment]
just like the advenced example.

we defined a mutation type in mutation_types
``` js
export const SET_TIP = 'SET_TIP'
```
and diapatch it in modules
``` js
import {
  SET_TIP
} from '../mutation_types'
const mutations = {
  [SET_TIP] (state, tip) {
    state.tip = tip
  }
}
```

if we forget to defined it in mutation_types or we just make a spelling mistake.

it won't throw an error, and it will change other states.

**Comment by ktsn** [comment]
Ah, I get it.
The problem here is Vuex accepts `undefined` type in dispatch time. Maybe v2.0 behaves as same.
We can assert dispatched type whether it is string value or not.

**Comment by ktsn** [comment]
Fixed via #552, #554
